src/unistd.rs

```diff
@@ -819,6 +819,64 @@ pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
     Ok((fd, PathBuf::from(pathname)))
 }
 
+libc_bitflags!{
+    pub flags AccessFlags: c_int {
+        R_OK,
+        W_OK,
+        X_OK,
+    }
+}
+
+pub enum AccessMode {
+    F_OK,
+    Flags(AccessFlags),
+}
+
+impl AccessMode {
+    fn to_libc_type(&self) -> c_int {
+        match self {
+            &AccessMode::F_OK => libc::F_OK,
+            &AccessMode::Flags(flags) => flags.bits(),
+        }
+    }
+}
+
+/// Check whether a file exists or whether the current process can access a file.
```

Comment:
Please link to the man page for this. See the `fork()` function doccomments for an example.